libbpfgo.go

```diff
@@ -239,23 +270,35 @@ const (
 	Tracing
 	XDP
 	Cgroup
+	CgroupLegacy
 	Netns
 )
 
 type BPFLink struct {
-	link      *C.struct_bpf_link
-	prog      *BPFProg
-	linkType  LinkType
-	eventName string
+	link       *C.struct_bpf_link
+	prog       *BPFProg
+	linkType   LinkType
+	eventName  string
+	attachType BPFAttachType // needed by legacy cgroup attachment
```

Comment:
Do you think we should change the name from `attachType` to something like "attachTypeLegacy" to note that this instead of a comment? To make it less confusing to implement APIs that create BPFLinks. (same comment for cgroupDir)

Also this got me confused about the difference between link/attach types (which I have before and then forgot again). I opened https://github.com/aquasecurity/libbpfgo/issues/217, just noting.

Reply:
> Also this got me confused about the difference between link/attach types (which I have before and then forgot again). I opened https://github.com/aquasecurity/libbpfgo/issues/217, just noting.

It makes sense. I would also use something like "attachTypeLegacy".

Reply:
I see your point... but BPFAttachType is unrelated to the "legacy" concept as well.

Every BPFLink should have its "attachType" set (and it does inside libbpf) IF libbpf wasn't able to detect the needed attachment type automatically (there is a table that relates SEC(), eBPF program types and their attachment capabilities).

Libbpfgo BPFLink interface is unaware of their attachment types, when instantiated, because of the interface libbpf provides... but it might be needed to know which attachment type that link used (or even influence which to use if multiple available).

NEVERTHELESS, I believe you have a valid point by "isolating" what is legacy from what is not and I did a change:

```go
type BPFLinkLegacy struct {
	attachType BPFAttachType
	cgroupDir  string
}

type BPFLink struct {
	link      *C.struct_bpf_link
	prog      *BPFProg
	linkType  LinkType
	eventName string
	legacy    *BPFLinkLegacy
}
```

already addressing your concern bellow (about link not existing).